Runtime/Scripts/PersistentConnection.cs

```diff
@@ -296,104 +327,176 @@ IEnumerator OnIntervalEdgeEventsLatency(EdgeEventsConnection connection, string
       latencyUpdatesCounter++;
       yield return StartCoroutine(OnIntervalEdgeEventsLatency(connection, host));
     }
 
-    async void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
+    void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
     {
       Logger.Log("Received event type: " + edgeEvent.EventType);
-      FindCloudletEventTrigger trigger = event_trigger_dict[edgeEvent.EventType];
-      if (config.newFindCloudletEventTriggers.Contains(trigger))
+      List<FindCloudletEventTrigger> fcTriggers = config.newFindCloudletEventTriggers;
+      switch (edgeEvent.EventType)
       {
-        switch (trigger)
-        {
-          case FindCloudletEventTrigger.Error:
-            PropagateError(FindCloudletEventTrigger.Error, edgeEvent.ErrorMsg);
-            return;
-          case FindCloudletEventTrigger.LatencyTooHigh:
-            if (edgeEvent.EventType == ServerEventType.EventLatencyRequest)
+        case ServerEventType.EventInitConnection:
+          Logger.Log("Successfully initiated persistent edge event connection");
+          return;
+        case ServerEventType.EventLatencyRequest:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh))
+          {
+            StartCoroutine(RespondToLatencyRequest());
+          }
+          return;
+        case ServerEventType.EventLatencyProcessed:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh))
+          {
+            latestServerStats = edgeEvent.Statistics;
+            if(!latencyProcessingInProgress)
             {
-              await RespondToLatencyRequest();
+              startLatencyProcessing = true;
+            }
+          }
+          return;
+        case ServerEventType.EventAppinstHealth:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.AppInstHealthChanged))
+          {
+            if (edgeEvent.NewCloudlet == null)
+            {
+              PropagateError(FindCloudletEventTrigger.AppInstHealthChanged, "Received EventAppinstHealth Event but NewCloudlet is null");
               return;
             }
-            ProcessLatency(edgeEvent.Statistics);
-            return;
-          default:
-            if (edgeEvent.NewCloudlet != null)
+            PropagateSuccess(FindCloudletEventTrigger.AppInstHealthChanged, edgeEvent.NewCloudlet);
+          }
+          return;
+        case ServerEventType.EventCloudletMaintenance:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.CloudletMaintenanceStateChanged))
+          {
+            if (edgeEvent.NewCloudlet == null)
             {
-              PropagateSuccess(trigger, edgeEvent.NewCloudlet);
+              PropagateError(FindCloudletEventTrigger.CloudletMaintenanceStateChanged, "Received CloudletStateChanged Event but NewCloudlet is null");
+              return;
             }
-            return;
-        }
+            PropagateSuccess(FindCloudletEventTrigger.CloudletStateChanged, edgeEvent.NewCloudlet);
```

Comment:
this should be  `PropagateSuccess(FindCloudletEventTrigger.CloudletMaintenanceStateChanged, edgeEvent.NewCloudlet);`